parse_kubeconfig.go

```diff
@@ -29,8 +29,9 @@ func KubeConfigToConnection(kubeconfig KubeConfig, inlineFiles bool) (Connection
 	}
 	var context *KubeConfigContext
 	for _, ctx := range kubeconfig.Contexts {
-		if ctx.Name == *kubeconfig.CurrentContext {
-			context = &ctx
+		ctxLocal := ctx
+		if ctxLocal.Name == *kubeconfig.CurrentContext {
+			context = &ctxLocal
```

Comment:
It's a tiny thing, but consider copying `ctx` only if we're gonna keep it:
```golang
		if ctx.Name == *kubeconfig.CurrentContext {
			ctxLocal := ctx
			context = &ctxLocal
```
Ditto for the subsequent changes.

Reply:
wouldn't that open `ctx.Name` up to being the wrong value?

Reply:
No:  the value of `ctx` is evaluated _during_ the loop body execution (it's not like the `t.Run()` case); the issue that we're fighting is that `ctx` will be reused for subsequent iterations and its will change, so we have to make sure that `context` points to something which contains the _value_ that we want...so we copy that value and point at the copy.

But, we don't need the copy if we're not going to point at it.